tests/unit-tests/gconstruct/test_construct_graph.py

```diff
@@ -1705,6 +1705,58 @@ def test_gc():
     assert not os.path.isdir("/tmp_featurewrapper2"), \
         "Directory /tmp_featurewrapper2 should not exist after gc"
 
+
+def test_homo():
+    # single node type and edge type input
```

Comment:
You also need to test add reverse edges.

Reply:
Refer to ```tests/end2end-tests/data_process/homogeneous_test.sh``` and ```tests/end2end-tests/data_process/check_homogeneous.py```

Reply:
Unit test can find the error earlier. If it is possible to write a unit test, you need to write a unit test.